code/obj/item/device/radio.dm

```diff
@@ -333,15 +333,19 @@ Green Wire: <a href='?src=\ref[src];wires=[WIRE_TRANSMIT]'>[src.wires & WIRE_TRA
 			R.hear_radio(M, messages, lang_id)
 
 	var/list/heard_flock = list() // heard by flockdrones/flockmind
-
-	// Don't let them monitor Syndie headsets. You can get the radio_brain bioeffect at the start of the round, basically.
-	if (src.protected_radio != 1 && isnull(src.traitorradio))
-		for (var/mob/living/L in radio_brains)
-			receive += L
-
-		for(var/mob/zoldorf/z in the_zoldorf)
-			if(z.client)
-				receive += z
+	var/datum/game_mode/conspiracy/N = ticker.mode
+	var/protected_frequency = null
+	if(istype(N))
+		protected_frequency = N.agent_radiofreq
+
+    // Don't let them monitor Syndie headsets. You can get the radio_brain bioeffect at the start of the round, basically.
+		if (src.protected_radio != 1 && isnull(src.traitorradio) && protected_frequency != display_freq )
```

Comment:
pretty sure this breaks all this existing functionality and only works on conspiracy mode

Reply:
On the newer commits I did, I tested and it compiles and works as intended for conspiracy; I checked on the nukie gamemode and nukie comms are still protected from radio_brain. I checked on gangs and gangs comms actually are not protected from radio_brain in current iteration or in the master branch (I think since the headset is grouped differently from traitorradio). Radio_brains can still parse departmental channels when I tested in extended and traitormode too. Let me know if I missed anything; sorry again this has been a mess.